Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs

```diff
@@ -94,16 +97,16 @@ public async ValueTask<TItem> UpdateAsync(TItem value, CancellationToken cancell
         {
             await Task.CompletedTask;
 
-            TItem item = Items.FirstOrDefault(i => i.Id == value.Id && i.PartitionKey == value.PartitionKey);
+            TItem item = Items.Values.FirstOrDefault(i => i.Id == value.Id && i.PartitionKey == value.PartitionKey);
 
             if (item is not null)
             {
-                Items.Remove(item);
-                Items.Add(value);
+                Items.TryRemove(value.Id, out TItem _);
```

Comment:
This could be simplified as:

```csharp
return Items[value.Id] = value;
```

Using the [built-in indexer](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.item?view=net-5.0), which will create or update accordingly.